L1Trigger/L1THGCal/test/HGCalBackendStage1ParameterExtractor.cc

```diff
@@ -0,0 +1,293 @@
+  std::vector<uint32_t> tc_coord_u = TCcoord_uv.getParameter<std::vector<uint32_t>>("TCu");
+  std::vector<uint32_t> tc_coord_v = TCcoord_uv.getParameter<std::vector<uint32_t>>("TCv");
+  if (tc_coord_u.size() != tc_coord_v.size())
+    throw cms::Exception("BadParameter") << "TCu and TCv vectors should be of same size";
+  for (size_t i = 0; i < tc_coord_u.size(); ++i)
+    tc_coord_uv_.emplace(std::make_pair(tc_coord_u.at(i), tc_coord_v.at(i)), i);
+
+  // Get truncation parameters
+  const edm::ParameterSet& truncationParamConfig =
+      conf.getParameterSet("BackendStage1Params").getParameterSet("truncation_parameters");
+  roz_min_ = truncationParamConfig.getParameter<double>("rozMin");
+  roz_max_ = truncationParamConfig.getParameter<double>("rozMax");
+  roz_bins_ = truncationParamConfig.getParameter<uint32_t>("rozBins");
+  max_tcs_per_bins_ = truncationParamConfig.getParameter<std::vector<uint32_t>>("maxTcsPerBin");
+  phi_edges_ = truncationParamConfig.getParameter<std::vector<double>>("phiSectorEdges");
+}
+
+HGCalBackendStage1ParameterExtractor::~HGCalBackendStage1ParameterExtractor() {}
+
+void HGCalBackendStage1ParameterExtractor::beginRun(const edm::Run& /*run*/, const edm::EventSetup& es) {
+  triggerGeometry_ = es.getHandle(triggerGeomToken_);
+
+  json outJSON;
+
+  // fill MetaData
+  outJSON["MetaData"]["CMSSWversion"] = cmssw_version_;
+  outJSON["MetaData"]["fpgaId"] = the_fpga_;
+
+  // fill geometry configuration
+  outJSON["TriggerGeometryConfig"]["TriggerGeometryName"] = trigger_geom_;
+  outJSON["TriggerGeometryConfig"]["DisconnectedLayers"] = disconnected_layers_;
+  outJSON["TriggerGeometryConfig"]["DisconnectedModules"] = disconnected_modules_;
+  outJSON["TriggerGeometryConfig"]["JsonMappingFile"] = json_mapping_file_;
+  outJSON["TriggerGeometryConfig"]["L1TLinksMapping"] = L1T_links_mapping_;
+  outJSON["TriggerGeometryConfig"]["L1TModulesMapping"] = L1T_modules_mapping_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorLinksPerModule"] = scintillator_links_per_module_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorModuleSize"] = scintillator_module_size_;
+  outJSON["TriggerGeometryConfig"]["ScintillatorTriggerCellSize"] = scintillator_trigger_cell_size_;
+
+  // fill truncation parameters
+  outJSON["TruncationConfig"]["rozMin"] = roz_min_;
+  outJSON["TruncationConfig"]["rozMax"] = roz_max_;
+  outJSON["TruncationConfig"]["rozBins"] = roz_bins_;
+  outJSON["TruncationConfig"]["maxTcsPerBin"] = max_tcs_per_bins_;
+  outJSON["TruncationConfig"]["phiSectorEdges"] = phi_edges_;
+
+  // fill trigger geometry
+  fillTriggerGeometry(outJSON);
+
+  // write out JSON file
+  std::ofstream outputfile(outJSONname_.c_str());
+  outputfile << std::setw(4) << outJSON << std::endl;
+}
+
+void HGCalBackendStage1ParameterExtractor::fillTriggerGeometry(json& json_file) {
+  trigger_map_set trigger_cells;
+
+  // retrieve valid trigger cells
+  std::cout << "Getting EE trigger cells\n";
```

Comment:
`cout`'s (here and below) could be removed

Reply:
done